### PR TITLE
[Needs SkyDB Cache] Improve registry errors

### LINKED ISSUE #10: Improve registry errors
<a href="https://github.com/m-cat"></img></a> **Issue by [m-cat](https://github.com/m-cat)**
_Friday Mar 26, 2021 at 11:54 GMT_
_Originally opened as https://github.com/NebulousLabs/skynet-js/pull/384_

----

----

_**[m-cat](https://github.com/m-cat)** included the following code: https://github.com/NebulousLabs/skynet-js/pull/384/commits_

### PR DESCRIPTION
**Pull request by m-cat**
Original PR: https://github.com/NebulousLabs/skynet-js/pull/384
Closes #10

### CODE REVIEW

**Comment by m-cat** [comment]
Intermittent test failures. Likely cause: trying to set a revision number that has just been set. If the load balancer takes us to a different server, that server may not have the latest revision and may accept a "taken" revision number. This should be solved by caching the revision number internally in SkyDB.

File: src/utils/error.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  // eslint-disable-next-line
+  err: any
+): Error {
+  /* istanbul ignore next */
+  if (!err) {
+    return new Error("Invalid response error");
+  }
+  /* istanbul ignore next */
+  if (!err.response) {
+    console.log(`Full error: ${err}`);
```
Comment (ro-tex) [inline src/utils/error.ts:17]: Should we be console-logging stuff here?

File: src/utils/error.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * @param err - The error from Axios.
+ * @returns - The extracted response error.
+ */
+export function extractResponseError(
+  // eslint-disable-next-line
+  err: any
+): Error {
+  /* istanbul ignore next */
+  if (!err) {
+    return new Error("Invalid response error");
```
Comment (ro-tex) [inline src/utils/error.ts:13]: This error message is a little confusing. I would interpret it as "unknown/unexpected error" and not as "we expected to see an error here but there is none".

**Review by ro-tex — APPROVED** [review]
LGTM.

**Comment by m-cat** [comment]
Closing as out-of-date.

**Pull request closed without merging**